e2e-tests/batch-mode-tests.sh

```diff
@@ -0,0 +1,108 @@
+​
+if [ -z "$(ls /tmp/Patient)" ]; then
+  echo 'NO PARQUET FILES FOUND IN /tmp/'
+  exit 0
+fi
+​
+current_path=`pwd`
+
+echo "COPYING PARQUET TOOLS JAR FILE TO ROOT"
+
+cp parquet-tools-1.11.1.jar ~/
+​
+cd ~/
+
+echo "Finding number of patients, encounters and obs in openmrs server"
+​
+mkdir /tmp/omrs
+​
+curl -L -X GET -u admin:Admin123 http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ >> /tmp/omrs/patients.json
+total_test_patients=$(jq '.total' /tmp/omrs/patients.json)
+echo "total_test_patients ---> $total_test_patients"
+​
+curl -L -X GET -u admin:Admin123 http://localhost:8099/openmrs/ws/fhir2/R4/Encounter/ >> /tmp/omrs/encounters.json
+total_test_encounters=$(jq '.total' /tmp/omrs/encounters.json)
+echo "total_test_encounters ---> $total_test_encounters"
+​
+curl -L -X GET -u admin:Admin123 http://localhost:8099/openmrs/ws/fhir2/R4/Observation/ >> /tmp/omrs/obs.json
+total_test_obs=$(jq '.total' /tmp/omrs/obs.json)
+echo "total_test_obs ---> $total_test_obs"
+​
+sudo rm -rf /tmp/omrs
+​
+echo "Counting number of patients, encounters and obs sinked to parquet files"
+​
+total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount /tmp/Patient/)
+echo "total_patients_streamed ---> $total_patients_streamed"
+​
+total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount /tmp/Encounter/)
+echo "total_encounters_streamed ---> $total_encounters_streamed"
+​
+total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount /tmp/Observation/)
+echo "total_obs_streamed ---> $total_obs_streamed"
+​
+if [[ ($total_patients_streamed == *$total_test_patients*) && ($total_encounters_streamed == *$total_test_encounters*) && ($total_obs_streamed == *$total_obs_encounters*)]] ;
+then
+  echo "BATCH MODE WITH PARQUET SINK EXECUTED SUCCESSFULLY"
+else
+  echo "BATCH MODE WITH PARQUET SINK TEST FAILED"
+fi
+​
+​
+cd $current_path
+​
+echo "BATCH MODE WITH FHIR SINK TESTS"
+​
+echo "STARTING FHIR SERVER"
+​
+docker-compose -f sink-compose.yml up -d
+fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi http://localhost:8098/fhir/Observation)
+until [ "$fhir_server_status_code" -eq 200 ]; do
+      sleep 20s;
+      echo 'WAITING FOR FHIR SERVER TO START'
+      fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi http://localhost:8098/fhir/Observation)
+done
+echo 'FHIR SERVER STARTED SUCCESSFULLY'
+​
+mvn exec:java -pl batch \
+ "-Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs --openmrsUserName=admin --openmrsPassword=Admin123 \
+   --searchList=Patient --batchSize=20 --targetParallelism=20 \
+   --fhirSinkPath=http://localhost:8098/fhir \
+   --sinkUserName=hapi --sinkPassword=hapi"
+
+mkdir /tmp/fhir
```

Comment:
For these too, I think it is better to use `mktemp` possibly with a `fhir` prefix to distinguish.